src/base/http/server.cpp

```diff
@@ -55,14 +55,14 @@ namespace
 
     QList<QSslCipher> safeCipherList()
     {
-        const QStringList badCiphers {u"idea"_qs, u"rc4"_qs};
+        const QStringList badCiphers {u"idea"_qs, u"rc4"_qs, u"rsa"_qs};
         const QList<QSslCipher> allCiphers {QSslConfiguration::supportedCiphers()};
         QList<QSslCipher> safeCiphers;
         std::copy_if(allCiphers.cbegin(), allCiphers.cend(), std::back_inserter(safeCiphers), [&badCiphers](const QSslCipher &cipher)
         {
             return std::none_of(badCiphers.cbegin(), badCiphers.cend(), [&cipher](const QString &badCipher)
             {
-                return cipher.name().contains(badCipher, Qt::CaseInsensitive);
+                return cipher.name().contains(badCipher, Qt::CaseInsensitive) || cipher.name().startsWith(u"dhe"_qs);
```

Comment:
See my new comment. This change is wrong. It disables the Diffie-Helman **Ephemeral** variant which is still considered good.